### ISSUE TITLE
On Windows 10 taskbar icon points to non-existent "chrome_proxy.exe" instead of "nw.exe".

### ISSUE THREAD

**Issue by LostBeard** [nwjs/nw.js#8227]
### Issue Type

- [X] Bug Report
- [X] Successfully reproduced against the latest version of NW.js

### Expected/Proposed Behavior

I expected the taskbar icon to point to the `nw.exe` app.

### Additional Info

- Operating System: Windows 10 x64
- NW.js Version: 0.92.0 sdk

### To reproduce
On a Windows 10 x64 PC (32bit has same issue)
- Download and extract [nwjs-sdk-v0.92.0-win-x64.zip](https://dl.nwjs.io/v0.92.0/nwjs-sdk-v0.92.0-win-x64.zip)
- Run `nw.exe` (same result with and without a manifest)
- Right click the NWJS taskbar icon 
- Right click the option: "NWJS Default"
- Click "Properties"
- The properties window will show that the icon `Target` is "...\chrome_proxy.exe" and not "...\nw.exe" as it should be.

This issue prevents the use of the "Pin to taskbar" option because the pinned icon will try to launch "chrome_proxy.exe" which does not exist.

This bug has been reported before as issue #7033. It was fixed and closed. This seems to be a regression due to changes since that fix.

The file that was modified to fix this issue before (#7033) was `chrome/browser/web_applications/components/web_app_shortcut_win.cc`. I wanted to look at that file to see what had changed and it seems it no longer exists. The folder `chrome/browser/web_applications/components` no longer exists ([web_applications](https://github.com/nwjs/chromium.src/tree/nw93/chrome/browser/web_applications)).

![NWjsPinnedTaskbarIconIssue](https://github.com/user-attachments/assets/b18326dd-07e4-4ca5-9133-3caf3426e583)

**Comment by LostBeard** [comment]
The files has moved to [chrome/browser/web_applications/os_integration/web_app_shortcut_win.cc](https://github.com/nwjs/chromium.src/blob/4a8af5225f5cd3176c04861578b8a6f228b6a92e/chrome/browser/web_applications/os_integration/web_app_shortcut_win.cc)

**Comment by ayushmanchhabra** [comment]
@rogerwang PR is ready for review

**Comment by ayushmanchhabra** [comment]
https://github.com/nwjs/chromium.src/pull/166
@rogerwang I have shared the PR URL for visibility, requesting your review. Thanks

**Comment by LostBeard** [comment]
Looking forward to this bug getting fixed. Is there an issue with the PR holding this up?

**Comment by LostBeard** [comment]
Any progress on this issue?

**Comment by LostBeard** [comment]
This bug is still present in version 0.95.

**Comment by rogerwang** [comment]
Will fix it in 0.96. Thanks.

**Comment by LostBeard** [comment]
Awesome! 

I tested 0.96 and it is fixed. Thank you!

## Issue side effect
Important note about a side effect caused by this issue that might affect others. This side effect can make it appear that the issue is not fixed even though it is.

### Previously with the bug
While using version 0.92 of NWjs which is affected by this issue, I had right clicked and pinned NWjs to the taskbar, which didn't work due to the bug. 

### Today with 0.96
I just tried 0.96 by downloading, unzipping and running NWjs. When I right clicked the taskbar icon for NWjs and clicked `Properties`, it was showing the properties information from the version of NWjs I had previously attempted to pin, including the incorrect shortcut to `chrome_proxy.exe`. I knew that folder no longer existed so it had to be getting the info from somewhere... It turns out Windows had created an `implicit app shortcut` every time I had tried pinning an NWjs exe. 

### Side effect fix
To fix, delete all the folders in `%AppData%\Roaming\Microsoft\Internet Explorer\Quick Launch\User Pinned\ImplicitAppShortcuts` that points to an NWjs exe that was affected by the issue.

![Image](https://github.com/user-attachments/assets/8731ef86-2d20-4681-8296-7d192c1946a5)

**Comment by LostBeard** [comment]
Not completely fixed. Now the target always points to `nw.exe` regardless of the actual exe name.

**Comment by LostBeard** [comment]
I do not believe NWjs uses a proxy exe like Chrome. Is there any reason why we cannot just return the executing exe path from the method `GetChromeProxyPath()` in the source file linked below? 

[chrome/browser/shortcuts/platform_util_win.cc](https://github.com/nwjs/chromium.src/blob/f4e83554086f6573ba9044f2bd11e2e6f24ed7ef/chrome/browser/shortcuts/platform_util_win.cc#L20)
```cpp
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/shortcuts/platform_util_win.h"

#include "base/check.h"
#include "base/files/file_path.h"
#include "base/path_service.h"

namespace shortcuts {

namespace {

constexpr base::FilePath::CharType kChromeProxyExecutable[] =
    FILE_PATH_LITERAL("nw.exe");

}  // namespace

base::FilePath GetChromeProxyPath() {
  base::FilePath chrome_dir;
  CHECK(base::PathService::Get(base::DIR_EXE, &chrome_dir));
  return chrome_dir.Append(kChromeProxyExecutable);
}

}  // namespace shortcuts
```

**Comment by LostBeard** [comment]
This is what I think should work. I'm pretty sure the build setup is a lot to get going. Anyone see an issue with this fix or test it? 

```cpp
base::FilePath GetChromeProxyPath() {
  base::FilePath chrome_exe;
  CHECK(base::PathService::Get(base::FILE_EXE, &chrome_exe));
  return chrome_exe;
}
```

**Comment by LostBeard** [comment]
My exact fix has been done before. 

https://github.com/nwjs/chromium.src/commit/80fd1a4ef4546c461bba40847a959418937be7d4

**Comment by LostBeard** [comment]
lol and it documented that is there to fix this issue. It must not have made it into 0.96?

Thank you @ayushmanchhabra 

Will it be in version 0.97?

**Comment by LostBeard** [comment]
It looks like you guys fixed it, and then broke it again.

This commit re-added the broken code that had been replaced with the code that fixed this issue.

https://github.com/nwjs/chromium.src/commit/3190d618befaa00e959e481abe06b9c4a57b82f6
